features/api-layer/dao-common/src/main/java/org/opennms/features/apilayer/dao/AlarmDaoImpl.java

```diff
@@ -83,4 +92,101 @@ public Optional<Alarm> getAlarmWithHighestSeverity(NodeRef nodeRef) {
+            });
+        });
+    }
+
+    @Override
+    public void acknowledge(final String user, final int... alarmIds) {
+        sessionUtils.withTransaction(() -> {
+            final List<OnmsAcknowledgment> acks = Arrays.stream(alarmIds).boxed().map(id -> {
+                final OnmsAcknowledgment ack = new OnmsAcknowledgment(user);
+                ack.setAckType(AckType.ALARM);
+                ack.setAckAction(AckAction.ACKNOWLEDGE);
+                ack.setRefId(id);
+                return ack;
+            }).collect(Collectors.toUnmodifiableList());
+            ackDao.processAcks(acks);
+        });
+    }
+
+    @Override
+    public void unacknowledge(final int... alarmIds) {
+        sessionUtils.withTransaction(() -> {
+            final List<OnmsAcknowledgment> acks = Arrays.stream(alarmIds).boxed().map(id -> {
+                final OnmsAcknowledgment ack = new OnmsAcknowledgment();
+                ack.setAckType(AckType.ALARM);
+                ack.setAckAction(AckAction.UNACKNOWLEDGE);
+                ack.setRefId(id);
+                return ack;
+            }).collect(Collectors.toUnmodifiableList());
+            ackDao.processAcks(acks);
+        });
+    }
+
+    @Override
+    public void escalate(final String user, final int... alarmIds) {
+        sessionUtils.withTransaction(() -> {
+            final List<OnmsAcknowledgment> acks = Arrays.stream(alarmIds).boxed().map(id -> {
+                final OnmsAcknowledgment ack = new OnmsAcknowledgment(user);
+                ack.setAckType(AckType.ALARM);
+                ack.setAckAction(AckAction.ESCALATE);
+                ack.setRefId(id);
+                return ack;
+            }).collect(Collectors.toUnmodifiableList());
+            ackDao.processAcks(acks);
+        });
+    }
+
+    @Override
+    public void clear(final int... alarmIds) {
+        sessionUtils.withTransaction(() -> {
+            final List<OnmsAcknowledgment> acks = Arrays.stream(alarmIds).boxed().map(id -> {
+                final OnmsAcknowledgment ack = new OnmsAcknowledgment();
+                ack.setAckType(AckType.ALARM);
+                ack.setAckAction(AckAction.CLEAR);
+                ack.setRefId(id);
+                return ack;
+            }).collect(Collectors.toUnmodifiableList());
+            ackDao.processAcks(acks);
+        });
+    }
+
+    @Override
+    public void setSeverity(final Severity severity, final int... alarmIds) {
+        final var onmsSeverity = ModelMappers.fromSeverity(severity);
+        sessionUtils.withTransaction(() -> {
+            final Criteria criteria = new CriteriaBuilder(OnmsAlarm.class)
+                    .and(Restrictions.in("id", Arrays.asList(alarmIds)))
+                    .toCriteria();
+            final List<OnmsAlarm> alarms = alarmDao.findMatching(criteria);
+            alarms.forEach(alarm -> {
+                alarm.setSeverity(onmsSeverity);
+                alarmDao.saveOrUpdate(alarm);
```

Comment:
Should also call `AlarmEntityNotifier#didUpdateAlarmSeverity` after `saveOrUpdate`.